src/backend/distributed/connection/placement_connection.c

```diff
@@ -346,6 +409,94 @@ CheckExistingConnections(uint32 flags, const char *userName,
 }
 
 
+static ConnectionReference *
+CheckExistingColocatedConnections(uint32 flags, const char *userName,
```

Comment:
The placement_connection.c versions don't do the conflict checks/errors, so they are a bit different than FindAvailableConnection.  I don't mind naming them similarly regardless though. What do you think?